lib/redis/namespace.rb

```diff
@@ -364,6 +364,15 @@ def call_with_namespace(command, *args, &block)
 
       (before, after) = handling
 
+      # Avoid modifying the caller's arguments.
+      args = args.map do |arg|
+        if arg.is_a?(Hash)
+          arg.dup
```

Comment:
It looks like add_namespace already duplicates the hash rather than modifying it in-place, so it looks like this is a bug is limited to certain command styles.  It looks like the affected command styles are:
* :sort
* :eval_style
* :scan_style

add_namespace also handles recursively nested hash and array arguments, so I'm not sure if we can even depend on this duplication to solve the problem elsewhere.  I think maybe the arguments should just be duplicated once rather than partially duplicating them in two places.